django/template/engine.py

```diff
@@ -29,7 +31,13 @@ def __init__(
         libraries=None,
         builtins=None,
         autoescape=True,
+        multiline=False,
     ):
+        if not multiline:
```

Comment:
I think we might want to do this differently as the only way folks can disable the deprecation warning is setting `'multiline': True,` in their settings. I would expect explicitly setting `'multiline': False,` should also silence the deprecation warning as their would be no change for them in upcoming releases?

If not, then I think multiline aims to be a transitional setting, which we are adding and immediately deprecating. I don't think the deprecation plan is completely clear to me

Reply:
Yea, the idea is that it's a setting that is immediately deprecated. The idea being that extremely few projects would break if the setting was just flipped, and the change is fairly easy to do if it does break due to this. So a fast track is the plan.

Reply:
Ok, based off other deprecation pathways:
- I don't think this should be added to new projects by default (as we don't do this with other transitional settings)
- I think perhaps make this a setting separate to the Templates setting as it sounds like we will remove the multiline argument and it will always support multiline. This setting is something folks can use to "opt-in" to using the new functionality in the 6.x cycle.

This also means less docs changes as the Engine signature won't change.
I also think it might be slightly easier for opting into the new format (to have no warnings) in the tests as you can do something like `@override_settings(TEMPLATE_TAGS_ASSUME_MULTILINE=True)`

What do you think?

Reply:
1. Why not? This is a bit of a special case where we think very few projects would be affected, so setting up new projects to use the future default seems like a good idea.
2. Hmm.. so like.. the deprecation warning and the flip between the two behaviors would not be engine creation time, but rendering time? I can get behind that. It certainly could make the testing smoother.

Reply:
On 1, we're adding something to the default template that's already deprecated and going to be removed. I think that's an issue

Reply:
Ah, yea ok, now I understand. Sure. If we move that to a `settings.TEMPLATE_TAGS_MULTILINE` or whatever, then projects will have a garbage setting lying around forever too, or we have to deprecate that and then people will have to remove it. Maybe that's better though...

Reply:
I looked into changing to use `settings.TEMPLATE_TAGS_MULTILINE` and it was a huge mess. The issue is that the engine is initialized in many places before django setup has been run. I spend an hour or so trying to make that all work, but I eventually gave up to get back to my day job.